test/uk/gov/hmrc/statepension/controllers/CopeErrorHandlingSpec.scala

```diff
@@ -0,0 +1,104 @@
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.libs.json.Json
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.test.UnitSpec
+import uk.gov.hmrc.http._
+import play.api.test.Helpers._
+import play.api.test.Injecting
+import play.api.inject.bind
+import uk.gov.hmrc.api.controllers.{ErrorGenericBadRequest, ErrorInternalServerError, ErrorNotFound}
+import uk.gov.hmrc.statepension.config.AppConfig
+import uk.gov.hmrc.statepension.controllers.ErrorResponses.ExclusionCopeProcessing
+
+import scala.concurrent.Future
+
+class CopeErrorHandlingSpec extends UnitSpec with GuiceOneAppPerSuite with Injecting with MockitoSugar {
+
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  override def fakeApplication(): Application = GuiceApplicationBuilder()
+    .configure("cope.feature.enabled" -> true).build()
+
+  val copeErrorHandling = inject[CopeErrorHandling]
+
+  "errorWrapper" must {
+    "return NotFound when NotFoundException is passed" in  {
+      val result = copeErrorHandling.errorWrapper(Future.failed(new NotFoundException("Not Found")))
+
+      status(result) shouldBe 404
+      contentAsJson(result) shouldBe Json.toJson(ErrorNotFound)
+    }
+
+    "return GateWayTimeout when GatewayTimeoutException is passed" in  {
+      val result = copeErrorHandling.errorWrapper(Future.failed(new GatewayTimeoutException("Gateway Timeout")))
+
+      status(result) shouldBe 504
+    }
+
+    "return BadGateway" when {
+      List(new BadGatewayException("BadGateway"),
+        new Upstream4xxResponse("4xx Response", 401, 502),
+        new Upstream5xxResponse("5xx Response", 500, 500)
+      ) foreach {
+        exception =>
+          s"${exception.getMessage} is passed" in  {
+            val result = copeErrorHandling.errorWrapper(Future.failed(exception))
+
+            status(result) shouldBe 502
+          }
+      }
+    }
+
+    "return Bad Request when BadRequestException is passed" in  {
+      val result = copeErrorHandling.errorWrapper(Future.failed(new BadRequestException("Bad Request")))
+
+      status(result) shouldBe 400
+      contentAsJson(result) shouldBe Json.toJson(ErrorGenericBadRequest("Upstream Bad Request. Is this customer below State Pension Age?"))
+    }
+
+    "return Forbidden" when {
+      "Upstream4xxResponse is returned with 422 status and NO_OPEN_COPE_WORK_ITEM message from DES" in  {
+        val result = copeErrorHandling.errorWrapper(Future.failed(new Upstream4xxResponse("NO_OPEN_COPE_WORK_ITEM", 422, 500)))
+
+        status(result) shouldBe 403
+        contentAsJson(result) shouldBe Json.toJson(ExclusionCopeProcessing(inject[AppConfig]))
+      }
+
+      "Upstream4xxResponse is returned with 422 status and CLOSED_COPE_WORK_ITEM message from DES" in  {
+        val result = copeErrorHandling.errorWrapper(Future.failed(new Upstream4xxResponse("CLOSED_COPE_WORK_ITEM", 422, 500)))
```

Comment:
Why 500? Should it be 422?

Reply:
After looking at how the rawReads returns a 422 response in Bootstrap. We see that it returns an Upstream4xxResponse that reports as a 500. We have mimicked this in the test. Here's the code.
https://github.com/hmrc/http-verbs/blob/eb80b3de0187317920cf23a287cfb81fc55052d3/http-verbs-common/src/main/scala/uk/gov/hmrc/http/HttpErrorFunctions.scala#L48